packages/commonwealth/client/scripts/state/api/comments/createComment.ts

```diff
@@ -0,0 +1,83 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import axios from 'axios';
+import { updateLastVisited } from 'controllers/app/login';
+import Comment from 'models/Comment';
+import app from 'state';
+import { ApiEndpoints } from 'state/api/config';
+import { updateThreadInAllCaches } from '../threads/helpers/cache';
+import useFetchCommentsQuery from './fetchComments';
+
+interface CreateCommentProps {
+  address: string;
+  threadId: number,
+  chainId: string;
+  unescapedText: string,
+  parentCommentId: number,
+  existingNumberOfComments: number
+}
+
+const createComment = async ({
+  chainId,
+  address,
+  threadId,
+  unescapedText,
+  parentCommentId = null,
+}: CreateCommentProps) => {
+  const {
+    session = null,
+    action = null,
+    hash = null,
+  } = await app.sessions.signComment({
+    thread_id: threadId,
+    body: unescapedText,
+    parent_comment_id: parentCommentId,
+  });
+
+  const response = await axios.post(
+    `${app.serverUrl()}/threads/${threadId}/comments`,
+    {
+      author_chain: chainId,
+      chain: chainId,
+      address: address,
+      parent_id: parentCommentId,
+      text: encodeURIComponent(unescapedText),
+      jwt: app.user.jwt,
+      canvas_action: action,
+      canvas_session: session,
+      canvas_hash: hash,
+    }
+  );
+
+  return new Comment(response.data.result)
+};
+
+const useCreateCommentMutation = ({ chainId, threadId, existingNumberOfComments = 0 }: Partial<CreateCommentProps>) => {
+  const queryClient = useQueryClient();
+  const { data: comments } = useFetchCommentsQuery({
+    chainId,
+    threadId,
+  })
+
+  return useMutation({
+    mutationFn: createComment,
+    onSuccess: async (newComment) => {
+      // update fetch comments query state
+      const key = [ApiEndpoints.FETCH_COMMENTS, chainId, threadId]
+      queryClient.cancelQueries({ queryKey: key });
+      queryClient.setQueryData(key,
+        () => {
+          return [...comments, newComment]
+        }
+      );
+
+      // TODO: these types of async calls should also have a dedicated react query handler
```

Comment:
maybe we can create ticket for this right away so it won't be TODO without action

Reply:
yeah good idea. @jnaviask I remember we had a convo somewhere related to [`updateLastVisited`](https://github.com/hicommonwealth/commonwealth/pull/4588/files/5701601dc1e7117e13ff3f95d10241579561a925#diff-5d40852891eaa12669f68e33089d0aa46c878619deef939df00f6f706fd89a20R75) and if we really need it.

Should we make a ticket to separate them into dedicated files or just remove them completely?

Reply:
@mzparacha please make a ticket with the full context + possible courses of action

Reply:
Ticket [here](https://github.com/hicommonwealth/commonwealth/issues/4701). I assigned us both.

cc: @jnaviask

Reply:
Ticket [here](https://github.com/hicommonwealth/commonwealth/issues/4701) @jnaviask